uui/components/pickers/DataPickerBody.tsx

```diff
@@ -49,6 +49,23 @@ export function DataPickerBody<TItem, TId>({ highlightSearchMatches = true, ...p
         if (e.shiftKey && e.key === 'Tab') e.preventDefault();
     };
 
+    const focusedRowId = useMemo((): string => {
+        // No need to make unnecessary calculations.
+        if (!props.showSearch) {
+            return '';
+        }
+
+        const focusedRow = props.rows.find((row) => {
```

Comment:
Another approach of searching focused items would be to introduce `focusedId` (and change `id` value in `DataPickerRow`) or `focusedRowKey` alongside `focusedIndex` to avoid O(n) complexity. Is making the API more complex worth it in this case, @AlekseyManetov?